editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts

```diff
@@ -86,66 +92,86 @@ export function setGridPropsCommands(
-    columnEnd != null &&
-    typeof columnEnd === 'number' &&
-    columnStart === columnEnd - 1
-  ) {
-    commands.push(
-      setProperty('always', elementPath, PP.create('style', 'gridColumn'), lineColumnStart),
-    )
-  } else {
-    if (columnStart != null) {
-      commands.push(
-        setProperty('always', elementPath, PP.create('style', 'gridColumnStart'), lineColumnStart),
-      )
+
+  function printPin(pin: GridPositionOrSpan, axis: 'row' | 'column'): string | number {
+    if (isGridSpan(pin)) {
+      return stringifyGridSpan(pin)
     }
-    if (columnEnd != null) {
-      commands.push(
-        setProperty('always', elementPath, PP.create('style', 'gridColumnEnd'), lineColumnEnd),
-      )
+    if (isCSSKeyword(pin)) {
+      return pin.value
     }
+    const tracks =
+      axis === 'column' ? gridTemplate.gridTemplateColumns : gridTemplate.gridTemplateRows
+    const maybeLineName =
+      tracks?.type === 'DIMENSIONS'
+        ? tracks.dimensions.find((_, index) => index + 1 === pin.numericalPosition)?.lineName
+        : null
+    if (maybeLineName != null) {
+      return maybeLineName
+    }
+    return pin.numericalPosition ?? 'auto'
   }
 
-  if (rowStart != null && rowStart === rowEnd) {
-    commands.push(setProperty('always', elementPath, PP.create('style', 'gridRow'), lineRowStart))
-  } else if (
-    rowStart != null &&
-    typeof rowStart === 'number' &&
-    rowEnd != null &&
-    typeof rowEnd === 'number' &&
-    rowStart === rowEnd - 1
-  ) {
-    commands.push(setProperty('always', elementPath, PP.create('style', 'gridRow'), lineRowStart))
-  } else {
-    if (rowStart != null) {
-      commands.push(
-        setProperty('always', elementPath, PP.create('style', 'gridRowStart'), lineRowStart),
-      )
+  function serializeAxis(
+    startPosition: GridPositionOrSpan,
+    endPosition: GridPositionOrSpan,
+    axis: 'row' | 'column',
+  ): {
+    property:
+      | 'gridColumn'
+      | 'gridColumnStart'
+      | 'gridColumnEnd'
+      | 'gridRow'
+      | 'gridRowStart'
+      | 'gridRowEnd'
+    value: string | number
+  } {
+    const startValue = printPin(startPosition, axis)
+    const endValue = printPin(endPosition, axis)
+
+    if (isAutoGridPin(startPosition) && !isAutoGridPin(endPosition)) {
+      return {
+        property: axis === 'column' ? 'gridColumnEnd' : 'gridRowEnd',
+        value: endValue,
+      }
     }
-    if (rowEnd != null) {
-      commands.push(
-        setProperty('always', elementPath, PP.create('style', 'gridRowEnd'), lineRowEnd),
-      )
+
+    if (
+      (isCSSKeyword(endPosition) && endPosition.value === 'auto') ||
+      (isGridPositionNumericValue(endPosition) &&
+        isGridPositionNumericValue(startPosition) &&
+        (endPosition.numericalPosition ?? 0) >= (startPosition.numericalPosition ?? 0) &&
+        (endPosition.numericalPosition ?? 0) - (startPosition.numericalPosition ?? 0) <= 1) ||
+      startValue === endValue
```

Comment:
could you move this to a helper function with names? I had to ask Cursor what this code does 

(btw here's what it wrote

```
Is the end position 'auto'?
Are both positions numeric, and is the end position either equal to or one greater than the start position?
Are the start and end values equal?
```

)

Reply:
```
// ... existing code ...
function shouldUseSimplifiedGridFormat(
  startPosition: GridPositionOrSpan,
  endPosition: GridPositionOrSpan,
  startValue: string | number,
  endValue: string | number,
): boolean {
  // Case 1: End position is 'auto'
  if (isCSSKeyword(endPosition) && endPosition.value === 'auto') {
    return true
  }

  // Case 2: Both positions are numeric and span only 1 cell
  if (isGridPositionNumericValue(startPosition) && isGridPositionNumericValue(endPosition)) {
    const start = startPosition.numericalPosition ?? 0
    const end = endPosition.numericalPosition ?? 0
    const isOneOrNoCellSpan = end - start <= 1 && end >= start
    if (isOneOrNoCellSpan) {
      return true
    }
  }

  // Case 3: Start and end values are the same
  return startValue === endValue
}

// Then in the original code:
if (shouldUseSimplifiedGridFormat(startPosition, endPosition, startValue, endValue)) {
  return {
    property: axis === 'column' ? 'gridColumn' : 'gridRow',
    value: startValue,
  }
}
// ... existing code ...
```

impressive suggestion from Cursor!